### PR TITLE
[POM] Add Sonar settings

### PR DESCRIPTION
**Pull request by SerhatG**

### CODE REVIEW

File: source/pom.xml [hunk 1]
```diff
@@ -36,6 +36,12 @@
     <java.version>17</java.version>
 
     <amazons3.version>2.20.56</amazons3.version>
+
+    <sonar.projectKey>aerius_file-server</sonar.projectKey>
+    <sonar.moduleKey>${project.artifactId}</sonar.moduleKey>
+    <sonar.organization>aerius</sonar.organization>
+    <sonar.host.url>https://sonarcloud.io</sonar.host.url>
```
Comment (BertScholten) [inline source/pom.xml:43]: Perhaps we could move these to the root pom?

File: source/pom.xml [hunk 2]
```diff
@@ -100,6 +106,14 @@
...
         <groupId>org.jacoco</groupId>
         <artifactId>jacoco-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>prepare-agent</id>
+            <goals>
+              <goal>prepare-agent</goal>
+            </goals>
+          </execution>
+        </executions>
```
Comment (BertScholten) [inline source/pom.xml:116]: Think this is specified in the root pom in `pluginManagement` and as such doesn't need to be included? Could be wrong though, didn't test it.

Comment (SerhatG) [reply source/pom.xml:116]: @Hilbrand did it for Register as well, so I thought if the master did.. but you are right I think.. Let me revert it and test it out.

File: source/pom.xml (hunk 1 above)
Comment (SerhatG) [reply source/pom.xml:43]: Yeah would work as well. @Hilbrand thoughts?

File: source/pom.xml [hunk 3]
```diff
@@ -36,6 +36,12 @@
     <java.version>17</java.version>
 
     <amazons3.version>2.20.56</amazons3.version>
+
+    <sonar.projectKey>aerius_file-server</sonar.projectKey>
+    <sonar.moduleKey>${project.artifactId}</sonar.moduleKey>
+    <sonar.organization>aerius</sonar.organization>
+    <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+    <sonar.java.pmd.reportPaths>${project.build.directory}/dependency-check-report-pmd.xml</sonar.java.pmd.reportPaths>
```
Comment (BertScholten) [inline source/pom.xml:44]: Not generating any pmd reports atm, so a bit pointless.

**Review by BertScholten — APPROVED** [review]
LGTM, can move the properties later if you want to get the build going ;)

**Pull request merged**